### DISCUSSION TITLE
Scaling asset up/down?

### DISCUSSION THREAD

**Discussion by bigrig2212** [donmccurdy/glTF-Transform#1006] [Q&A] [answered]
Haven't seen an api for Scaling. Assuming there's no api for that?

**Comment by bigrig2212** [comment]
Oh, actually i see: https://github.com/donmccurdy/glTF-Transform/blob/main/packages/core/src/properties/node.ts
Hm, how to call that via the CLI and apply to the entire asset?

**Comment by donmccurdy** [comment, marked as answer]
Hi @bigrig2212! If you mean to rescale all or part of the scene (as opposed to textures) then no, the CLI does not provide commands for that. However, this can be done in a script. Related functions:

- [`Node#setScale`](https://gltf-transform.dev/modules/core/classes/Node#setScale)
- [`transformMesh`](https://gltf-transform.dev/modules/functions/functions/transformMesh)
- [`transformPrimitive`](https://gltf-transform.dev/modules/functions/functions/transformPrimitive)

A basic example, scaling scenes up by 2x, would look something like this:

```js
import { NodeIO } from '@gltf-transform/core';

const io = new NodeIO();
const document = await io.read('path/to/model.glb');

for (const scene of document.getRoot().listScenes()) {
  for (const node of scene.listChildren()) {
    const scale = node.getScale();
    scale[0] *= 2;
    scale[1] *= 2;
    scale[2] *= 2;
    node.setScale(scale);
  }
}

await io.write('path/to/model.glb', document);
```

**Reply by bigrig2212** [reply]
Ah k - thanks. Appreciate it. (have been doing batches from a shell script using the CLI... sounds like i'll need to write a different harness in node.)
